driver/configuration/provider_koanf.go

```diff
@@ -306,7 +346,11 @@ func (v *KoanfProvider) PipelineConfig(prefix, id string, override json.RawMessa
 		return errors.WithStack(err)
 	}
```

Comment:
The culprit of the high CPU usage is in this method. My recommendation would be to cache the result of the validation (only when successful) and the cache key `sha256(prefix + "|" + id + "|" + marshalled)`. That way we will respect hot reloading (because we use the config values) and still save a lot of CPU cycles!